### PR TITLE
fix metrics workqueues

### PR DESCRIPTION
**Pull request by tehlers320**
**What this PR does / why we need it**:
The metrics workqueues seem to appear randomly for various builds/users. This PR ensures the correct package always loads first. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # https://github.com/kubernetes-sigs/kubefed/issues/1372

**Special notes for your reviewer**:
ExpediaInc is in the process of getting the CLA signed ASAP.

### CODE REVIEW

File: cmd/controller-manager/main.go [hunk 1]
```diff
@@ -41,3 +41,7 @@ func main() {
 		os.Exit(1) //nolint:gocritic
 	}
 }
+
+// Without init, the _ imports do not initialize before app package
```
Comment (mars1024) [inline cmd/controller-manager/main.go:45]: Could you help to link the related golang specification about this behavior? Very curious about this.

Comment (tehlers320) [reply cmd/controller-manager/main.go:45]: @mars1024 i couldnt find a good doc on this, this is the only thing i could find https://golang.org/doc/effective_go#blank_import

```
For example, during its init function, the net/http/pprof package registers HTTP handlers that provide debugging information.
```

This implies that init is needed for a side effect import, though it does not clearly state it.

I've tested it extensively and it is fixing it.

Comment (tehlers320) [reply cmd/controller-manager/main.go:45]: Should i re-phrase the comment 
```
// init for _ imports 
```
instead ?

**Comment by mars1024** [comment]
Hi, @tehlers320 , I tested this locally but ran into an unexpected result, 

First run,
```
package main

import (
	_ "bruce/dep2"

	"bruce/dep1"
)

func init() {}

func main() {
	dep1.NoOp()
}
```
Result is 
```
dep2 init
dep1 init
```

Second run,
```
package main

import (
	_ "bruce/dep2"

	"bruce/dep1"
)

func main() {
	dep1.NoOp()
}
```

Result is still, 
```
dep2 init
dep1 init
```

What did I miss?

**Comment by tehlers320** [comment]
> What did I miss?

Sorry i didn't have time to dig into this and wont probably until the weekend, i did re-run your example test and i'm seeing the same thing. This is loading as expected with or without the func init() using example code.  I started drafting an e2e test to get to the bottom of this and will try and finish that ASAP to prove/disprove the issue.

**Comment by hectorj2f** [comment]
@mars1024 @tehlers320 What is the state of this PR? Is it working for you both?

**Comment by mars1024** [comment]
> @mars1024 @tehlers320 What is the state of this PR? Is it working for you both?

Sorry, all I know is already here, let's wait for more input from @tehlers320

**Comment by tehlers320** [comment]
> @mars1024 @tehlers320 What is the state of this PR? Is it working for you both?

It works, but it seems my co-worker has the proper fix. closing.

**Pull request closed without merging**